tests/test_proc/test_adaptive_spectral_filter.py

```
@@ -0,0 +1,783 @@
+    _get_engine,
+    _validate_window_and_overlap,
+)
+from dascore.utils.signal import _triangular_taper
+
+
+def _patch(
+    shape: tuple[int, ...],
+    dims: tuple[str, ...],
+    *,
+    dtype=np.float32,
+    time_step=np.timedelta64(4, "ms"),
```

Ruff (0.15.12)

[warning] Do not perform function call `np.timedelta64` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)